acceptance/tests/failover/timeout_failover.rb

```diff
@@ -0,0 +1,52 @@
+require 'pxp-agent/config_helper.rb'
+require 'pxp-agent/test_helper.rb'
+
+test_name 'C97964 - agent should use next broker if primary is timing out' do
+
+  PRIMARY_BROKER_INSTANCE = 0
+  REPLICA_BROKER_INSTANCE = 1
+  teardown do
+    kill_all_pcp_brokers(master)
+    run_pcp_broker(master,    PRIMARY_BROKER_INSTANCE)
+  end
+
+  step 'Ensure each agent host has pxp-agent configured with multiple uris, running and associated' do
+    agents.each do |agent|
+      on agent, puppet('resource service pxp-agent ensure=stopped')
+      num_brokers = 2
+      pxp_config = pxp_config_hash_using_puppet_certs(master, agent, num_brokers)
+      pxp_config['allowed-keepalive-timeouts'] = 0
+      create_remote_file(agent, pxp_agent_config_file(agent), pxp_config.to_json.to_s)
+      on(agent, "rm -rf #{logfile(agent)}")
+      on agent, puppet('resource service pxp-agent ensure=running')
+      show_pcp_logs_on_failure do
+        assert_equal(master[:pcp_broker_instance], PRIMARY_BROKER_INSTANCE, "broker instance is not set correctly: #{master[:pcp_broker_instance]}")
+        assert(is_associated?(master, "pcp://#{agent}/agent"),
+               "Agent identity pcp://#{agent}/agent for agent host #{agent} does not appear in pcp-broker's (#{broker_ws_uri(master)}) client inventory after ~#{PCP_INVENTORY_RETRIES} seconds")
+      end
+    end
+  end
+
+  step 'Block primary broker, start replica' do
+    block_pcp_broker(master,PRIMARY_BROKER_INSTANCE)
+    run_pcp_broker(master,  REPLICA_BROKER_INSTANCE)
+  end
+
+  step 'On each agent, test that a new association has occurred' do
+    assert_equal(master[:pcp_broker_instance], REPLICA_BROKER_INSTANCE, "broker instance is not set correctly: #{master[:pcp_broker_instance]}")
+    agents.each do |agent|
+      assert(is_associated?(master, "pcp://#{agent}/agent"),
```

Comment:
In terms of how much time the test allows for failover to occur; I think this line imposes around 60 seconds (after the broker appears to be up) for the agent to failover or else we fail - is that long enough? e.g. if it takes 90 seconds or 120 seconds, do we want this test to show as failed?